src/main/java/net/adoptium/UserSystem.java

```diff
@@ -0,0 +1,30 @@
+package net.adoptium;
+
+import net.adoptopenjdk.api.v3.models.Architecture;
+import net.adoptopenjdk.api.v3.models.OperatingSystem;
+
+public class UserSystem {
+    private OperatingSystem os;
+    private Architecture arch;
+
+    public void UserSystem(OperatingSystem os, Architecture arch) {
+        this.os = os;
+        this.arch = arch;
+    }
+
+    public void setOs(OperatingSystem os) {
```

Comment:
Does it really makes sense to change OS/Arch after object creation? Maybe it would be better to define the fields as final and remove the setter